src/core/thread/mesh_forwarder_ftd.cpp

```diff
@@ -235,20 +235,37 @@ void MeshForwarder::UpdateIndirectMessages(void)
     }
 }
 
-otError MeshForwarder::EvictIndirectMessage(void)
+otError MeshForwarder::EvictMessage(uint8_t aPriority)
 {
-    otError error = OT_ERROR_NOT_FOUND;
+    otError  error = OT_ERROR_NOT_FOUND;
+    Message *message;
 
-    for (Message *message = mSendQueue.GetHead(); message; message = message->GetNext())
-    {
-        if (!message->IsChildPending())
-        {
-            continue;
-        }
+    VerifyOrExit((message = mSendQueue.GetTail()) != NULL);
 
+    if (message->GetPriority() > aPriority)
+    {
         RemoveMessage(*message);
         ExitNow(error = OT_ERROR_NONE);
     }
+    else
+    {
+        while (1)
+        {
+            for (message = mSendQueue.GetHeadForPriority(aPriority); message && (message->GetPriority() == aPriority);
+                 message = message->GetNext())
+            {
+                if (message->IsChildPending())
+                {
+                    RemoveMessage(*message);
+                    ExitNow(error = OT_ERROR_NONE);
+                }
+            }
+
+            VerifyOrExit(aPriority != Message::kPriorityHigh);
+
+            aPriority--;
```

Comment:
Does decrementing `aPriority` here mean that we are looking for a higher priority message to evict?

Reply:
Yes. We are looking for higher priority child pending message to evict.